aten/src/ATen/native/mps/operations/Copy.mm

```diff
@@ -89,12 +89,12 @@ static void copy_cast_mps(at::Tensor& dst,
   Tensor dst = dst_;
   Tensor src = src_;
 
-  if (!dst_.is_contiguous(MemoryFormat::Contiguous) && !sameMemFormat) {
+  if (dst_.strides() != src_.strides()) {
```

Comment:
That looks suspicious? 
Anything that has stride but not storage_offset feels suspicious to me :D

Reply:
Copying from a MPS buffer to a CPU buffer should be done using a single blit copy, without making an extra contiguous buffer, copy into that, then scatter back in the original CPU buffer